webdictionary/src/main/java/com/eis0/kademlianetwork/SMSKademliaListener.java

```diff
@@ -108,7 +108,138 @@ public void onMessageReceived(SMSMessage message) {
                 SMSKademliaNode nodeToAdd = new SMSKademliaNode(idFound);
                 kadNet.getLocalRoutingTable().insert(nodeToAdd);
                 break;
-                
+
+
+            /**Joining a network */
+            case JoinPermission:
+                ConnectionHandler.sendAcceptRequest(peer);
+                break;
+            case AcceptJoin:
+                ConnectionHandler.acceptRequest(peer);
+                break;
+            case FindId:
+                //1. Creates the response addressed to the sender, to inform him of my activity
+                //status, that is that I'm alive and still active inside the network
+                sendAcknowledge(peer);
+                //2. Processes the information brought by the message received
+                Log.i("CONN_LOG", "IdFound: " + splitted[1]);
+                idToFind = new KademliaId(splitted[1]);
```

Comment:
Mh, mooving it elsewhere could be a good idea, it will be part of our restoration process (which we already started for all our code) because we noticed (and someone else made us notice) how we stacked too many functionalities in the same spots. About the hardwiring, the position is not always related to the same type of data, for example: sometimes the second value is the searcher peer, sometimes is the resource; but yes, it could be avoided/solved mooving the process elsewhere, even if it would require a different handling mechanism based upon the type of task to accomplish. Because, as I said, the structure of the incoming message changes upon the RequestType; it was an easier solution handling in loco the situation, maybe not the most modular. It will require some time, but it could be a good way to improve the ability of the system to process more complex informations

Reply:
> About the hardwiring, the position is not always related to the same type of data, for example: sometimes the second value is the searcher peer, sometimes is the resource; but yes, it could be avoided/solved mooving the process elsewhere, even if it would require a different handling mechanism based upon the type of task to accomplish. Because, as I said, the structure of the incoming message changes upon the RequestType; it was an easier solution handling in loco the situation, maybe not the most modular.

`sometimes the second value is the searcher peer, sometimes is the resource` Ok I didn't see that, I must say it's even worse. I understand the practicality, but if in the future you will need to change this code, you'll find it very difficult to understand. Too many cases.
I suggest that you and your team decide a fixed syntax, and add it in the java docs. I think you shouldn't let pass too much time, as you'll find it more difficult to fix it.